src/joy_api_test.c

```diff
@@ -88,23 +87,142 @@ int process_pcap_file (unsigned long index, char *file_name) {
+        } else {
+           printf("tls version=unknown\n");
+        }
+    }
+}
+
+void my_splt_callback(void *curr_rec, unsigned int data_len, unsigned char *data) {
+    unsigned int i = 0;
+    unsigned int splt_recs = 0;
+    short *formatted_data = (short*)data;
+
+    if ((curr_rec == NULL) || (formatted_data == NULL)) return;
+
+    splt_recs = data_len / 4;
+    if (splt_recs > 10) {
+        printf("Incorrect SPLT Data Length (%d)\n",data_len);
+        return;
+    }
+
+    printf("SPLT REC(%d) LENGTHS: ",splt_recs);
+    for (i=0; i<splt_recs; ++i) {
+        if (i == (splt_recs-1))
+            printf("{%d}", *(formatted_data+i));
+        else
+            printf("{%d}, ", *(formatted_data+i));
+    }
+    printf("\n");
+    printf("SPLT REC(%d) TIMES: ",splt_recs);
+    for (i=0; i<splt_recs; ++i) {
+        if (i == (splt_recs-1))
+            printf("{%d}", *(formatted_data+splt_recs+i));
+        else
+            printf("{%d}, ", *(formatted_data+splt_recs+i));
+    }
+    printf("\n");
+}
+
+void my_salt_callback(void *curr_rec, unsigned int data_len, unsigned char *data) {
+    unsigned int i = 0;
+    unsigned int salt_recs = 0;
+    short *formatted_data = (short*)data;
+
+    if ((curr_rec == NULL) || (formatted_data == NULL)) return;
+
+    salt_recs = data_len / 4;
+    if (salt_recs > 10) {
+        printf("Incorrect SALT Data Length (%d)\n",data_len);
+        return;
+    }
+
+    printf("SALT REC(%d) LENGTHS: ",salt_recs);
+    for (i=0; i<salt_recs; ++i) {
+        if (i == (salt_recs-1))
+            printf("{%d}", *(formatted_data+i));
+        else
+            printf("{%d}, ", *(formatted_data+i));
+    }
+    printf("\n");
+    printf("SALT REC(%d) TIMES: ",salt_recs);
+    for (i=0; i<salt_recs; ++i) {
+        if (i == (salt_recs-1))
+            printf("{%d}", *(formatted_data+salt_recs+i));
+        else
+            printf("{%d}, ", *(formatted_data+salt_recs+i));
+    }
+    printf("\n");
+}
+
+void my_bd_callback(void *curr_rec, unsigned int data_len, unsigned char *data) {
+    int i = 0;
+    uint16_t *formatted_data = (uint16_t*)data;
+
+    if ((curr_rec == NULL) || (formatted_data == NULL)) return;
+
+    if (data_len != 512) return;
+
+    /* Each ASCII byte value */
+    printf("BYTE COUNTS: ");
+    for (i=0; i<256; ++i) {
+        if (i == 255)
+            printf("{%d}", *(formatted_data+i));
+        else
+            printf("{%d}, ", *(formatted_data+i));
+    }
+    printf("\n");
+}
+
 void *thread_main1 (void *file)
 {
+    unsigned int recs = 0;
+
     sleep(1);
     printf("Thread 1 Starting\n");
     joy_print_config(0, JOY_JSON_FORMAT);
-    process_pcap_file(0, file);
-    joy_print_flow_data(0, JOY_ALL_FLOWS);
+    if (file != NULL) {
+        proc_pcap_file(0, file);
+        joy_idp_external_processing(0, my_idp_callback);
+        joy_tls_external_processing(0, my_tls_callback);
+        //joy_splt_external_processing(0, JOY_NFV9_EXPORT, 1, my_splt_callback);
```

Comment:
Remove dead code. Commented out code.

Reply:
This is my test API tool and I keep around various API calls to easily hit different functions and features. This way I don't have to re-code them, I can just comment out and uncomment others to hit different functions.

I can remove them if it really bothers you.